lib/std/stropt.nim

```diff
@@ -0,0 +1,71 @@
+#
+#
+#           The Nim Compiler
+#        (c) Copyright 2021 Nim Contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+const whitespaces = {' ', '\t', '\v', '\r', '\l', '\f'}
+
+func stripSlice*(s: openArray[char], leading = true, trailing = true, chars: set[char] = whitespaces): Slice[int] =
+  runnableExamples:
+    doAssert stripSlice(" abc  ") == 1 .. 3
+  var
+    first = 0
+    last = high(s)
+  if leading:
+    while first <= last and s[first] in chars: inc(first)
+  if trailing:
+    while last >= first and s[last] in chars: dec(last)
+  result = first .. last
+
+func setSlice*(s: var string, slice: Slice[int]) =
+  ## Inplace version for `substr`.
+  let first = slice.a
+  let last = slice.b
+  if first > last:
+    s.setLen(0)
+    return
+  template impl = 
+    for index in first .. last:
+      s[index - first] = s[index]
+  if first > 0:
+    when nimvm: impl()
+    else:
+      # not JS and not Nimscript
+      when not declared(moveMem):
+        impl()
```

Comment:
in RT js, is `splice` more efficient?

see `{.emit: "`x`.splice(`i`, 1);".}` usd in system. delete